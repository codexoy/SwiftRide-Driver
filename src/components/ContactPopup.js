import React from "react"
import {
	StyleSheet,
	View,
	Text,
	TouchableHighlight,
	Linking
} from "react-native"
import Modal from "react-native-modal"
import { strings } from "../i18n"

export default class ContactPopup extends React.Component {
	constructor(props) {
		super(props)
	}

	call = () => {
		Linking.openURL(`tel:${this.props.phone}`)
	}

	whatsapp = () => {
		const phone = this.props.phone.substring(1) // remove leading 0
		Linking.openURL(`http://api.whatsapp.com/send?phone=972${phone}`)
	}

	render() {
		return (
			<Modal
				isVisible={this.props.visible}
				onBackdropPress={() => this.props.onPress()}
				animationIn="pulse"
				animationOut="fadeOut"
			>
				<View style={styles.popup} testID={this.props.testID}>
					<TouchableHighlight
						underlayColor="#f8f8f8"
						style={styles.row}
						onPress={this.whatsapp}
					>
						<Text style={styles.title}>{strings("whatsapp")}</Text>
					</TouchableHighlight>
					<TouchableHighlight
						underlayColor="#f8f8f8"
						style={styles.row}
						onPress={this.call}
					>
						<Text style={styles.title}>
							{strings("call_phone")}
						</Text>
					</TouchableHighlight>
				</View>
			</Modal>
		)
	}
}

const styles = StyleSheet.create({
	popup: {
		flex: 1,
		maxHeight: 120,
		backgroundColor: "#fff",
		alignSelf: "center",
		width: 320,
		alignContent: "center",
		borderRadius: 4
	},
	row: {
		width: "100%",
		alignItems: "center",
		justifyContent: "center",
		height: 60
	},
	title: {
		fontSize: 18,
		fontWeight: "bold"
	}
})
