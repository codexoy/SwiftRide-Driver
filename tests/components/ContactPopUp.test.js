import React from "react"

import renderer from "react-test-renderer"

import ContactPopup from "../../src/components/ContactPopUp"

describe("ContactPopup", () => {
	test("view renders correctly", () => {
		const phone = "0542224441"
		const tree = renderer
			.create(
				<ContactPopup
					visible={true}
					phone={phone}
					onPress={jest.fn()}
					testID="test"
				/>
			)
			.toJSON()
		expect(tree).toMatchSnapshot()
	})
})
