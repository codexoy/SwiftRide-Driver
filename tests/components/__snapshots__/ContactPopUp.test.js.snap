// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactPopup view renders correctly 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[Function]}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    hideModalContentWhileAnimating={false}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    pointerEvents="box-none"
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "margin": 37.5,
        "transform": Array [
          Object {
            "scale": 1,
          },
        ],
      }
    }
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
  >
    <View
      style={
        Object {
          "alignContent": "center",
          "alignSelf": "center",
          "backgroundColor": "#fff",
          "borderRadius": 4,
          "flex": 1,
          "maxHeight": 120,
          "width": 320,
        }
      }
      testID="test"
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "height": 60,
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 18,
              "fontWeight": "bold",
            }
          }
        >
          שלח WhatsApp
        </Text>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "height": 60,
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 18,
              "fontWeight": "bold",
            }
          }
        >
          התקשר
        </Text>
      </View>
    </View>
  </View>
</Modal>
`;
